controllers/openshift_controller.go

```diff
@@ -106,19 +105,40 @@ func (r *KataConfigOpenShiftReconciler) Reconcile(ctx context.Context, req ctrl.
 	}()
 }
 
-func (r *KataConfigOpenShiftReconciler) newMCPforCR() *mcfgv1.MachineConfigPool {
+func (r *KataConfigOpenShiftReconciler) newMCPforCR() (*mcfgv1.MachineConfigPool, error) {
 	lsr := metav1.LabelSelectorRequirement{
 		Key:      "machineconfiguration.openshift.io/role",
 		Operator: metav1.LabelSelectorOpIn,
-		Values:   []string{"kata-oc", "worker"},
+		Values:   []string{"worker", "kata-oc"},
 	}
 
-	var nodeSelector *metav1.LabelSelector
+	// Default NodeSelector is all machines with worker label.
+	// Otherwise all nodes including the control plane nodes will be put under new MCP
+	nodeSelector := &metav1.LabelSelector{MatchLabels: map[string]string{"node-role.kubernetes.io/worker": ""}}
+
+	if r.kataConfig.Spec.CheckNodeReadiness {
+		nodeSelector = metav1.AddLabelToSelector(nodeSelector, "feature.node.kubernetes.io/runtime.kata", "true")
+	}
 
 	if r.kataConfig.Spec.KataConfigPoolSelector != nil {
-		nodeSelector = r.kataConfig.Spec.KataConfigPoolSelector
+		// KataConfigPoolSelector can be a MatchExpression or MatchLabel. Need to Convert MatchExpression to MatchLabel
+		// and add it to nodeSelector
```

Comment:
I have used the standard helpers for conversion and re-arranged the existing code. I'm not sure if there are corner cases when this is not possible :-(. Any suggestions ?

Reply:
@pmores so the comment in the helper code mentions the case where it'll error out - https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/helpers.go#L74-L78

Since we are handling the error, I think it should be fine. Additionally we can add some failure cases in the documentation for now .